File: AGENDLY_APP/lib/features/auth/infrastructure/mappers/user_mapper.dart
import 'package:agendly_app/features/auth/domain/entities/user.dart';

class UserMapper {
  static User fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'] ?? json['_id'] ?? '',
      email: json['email'] ?? '',
      name: json['name'] ?? '',
      role: json['role'] ?? 'USER',
      businessId: json['businessId'],
      phone: json['phone'],
      avatarUrl: json['avatarUrl'],
      isActive: json['isActive'] ?? true,
    );
  }

  static Map<String, dynamic> toJson(User user) {
    return {
      'id': user.id,
      'email': user.email,
      'name': user.name,
      'role': user.role,
      'businessId': user.businessId,
      'phone': user.phone,
      'avatarUrl': user.avatarUrl,
      'isActive': user.isActive,
    };
  }
}

File: BACK/src/infrastructure/mappers/user.mapper.ts
import { UserEntity } from '../../domain/entities/user.entity';
import { IUser } from '../datasources/mongo/models/user.model';

export class UserMapper {
  static fromMongo(document: IUser): UserEntity {
    return {
      id: document._id.toString(),
      businessId: document.businessId,
      email: document.email,
      password: document.password,
      name: document.name,
      role: document.role as UserEntity['role'],
      fcmToken: document.fcmToken,
      avatarUrl: document.avatarUrl,
      isActive: document.isActive,
      createdAt: document.createdAt,
      updatedAt: document.updatedAt,
    };
  }
}
